roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala

```diff
@@ -158,6 +260,17 @@ object rblstring {
       Right(elem(n).asInstanceOf[Fixnum])
     }
 
+  /**
+    * Check the specified parameter for type Char. Return a PrimError if it is
+    * not else return the Char.
+    */
+  private def checkChar(n: Int, elem: Seq[Ob]): Either[PrimError, RblChar] =
```

Comment:
These `check` methods look quite repetitive, so I personally would prefer to replace them with some generic method, like:
```scala
def checkType[T <: Ob](n: Int, elem: Seq[Ob])(implicit ct: ClassTag[T]): Either[PrimError,T] = {
  elem(n) match {
    case e: T => Right(elem(n).asInstanceOf[T])
    case _ => Left(TypeMismatch(n, ct.runtimeClass.getName))
  }
}
```

Reply:
Done